src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/Agent/AgentOrchestrator.cs

```diff
@@ -0,0 +1,97 @@
+using Microsoft.Extensions.Logging;
+using Azure.Tools.ErrorAnalyzers;
+using Azure.Tools.GeneratorAgent.Configuration;
+
+namespace Azure.Tools.GeneratorAgent.Agent
+{
+    internal class AgentOrchestrator
```

Comment:
I actually think I liked ErrorFixerAgent more :sweat_smile: , because I think I would expect AgentOrchestrator to be orchestrating a bunch of distributed agents? But if I'm understanding correctly, this is more encapsulating one single agent and all the work needed to manage and interact with it?